docs/core/common/assets/js/api/core/ui-states.md

```diff
@@ -0,0 +1,178 @@
+		- Dispatches a custom event to the scope elements ( `e-ui-state:${ stateID }`, e.g. `e-ui-state:document/direction-mode` ) with `oldValue` and `newValue` under `e.detail`.
+	- Setting an invalid option to a state will throw an error.
+
+## Guidelines, Conventions & Files Structure:
+* Each UI state is owned by a [component](./components.md#guidelines-conventions--files-structure).
+* Each [component](./components.md#guidelines-conventions--files-structure) can override the `defaultUiStates()` method to return a set of UI states of its own.
+* The UI states are imported via built-in method called `importUiStates()`.
+	* All the UI states should be exported in one index file:
+	    ```javascript
+		// ui-states/index.js
+		export { DirectionMode } from './direction-mode';
+		export { ColorPickingMode } from './color-picking-mode';
+		```
+	  You can have as many indexes as you wish in every hierarchy under `component/ui-states/some/long/path` in order to organize your code, but the requirement is to have a single index file
+	  at `component/ui-states/index.js` which exports all the UI states. Take the **index.js** example above as a scenario:
+	    ```html
+		📦 component
+		│   📜 component.js
+		│
+		└───📂 ui-states
+		       📜 color-picking-mode.js
+		       📜 direction-mode.js
+		       📜 index.js ( has all the UI states exported )
+		```
+	* The UI states are being imported using `importUiStates()`, for example:
+	    ```javascript
+		import * as uiStates from './ui-states';
+
+		export class Component extends $e.modules.ComponentBase {
+			getNamespace() {
+				return 'component-name';
+			}
+
+			defaultUiStates() {
+				return this.importUiStates( uiStates );
+			}
+		}
+		```
+* UI State conventions:
+	```javascript
+	import UiStateBase from 'elementor-api/core/states/ui-state-base';
+	
+	export class {FILE_NAME_CAMEL_CASE} extends UiStateBase {
+		getId() {
+			// A unique ID for the current namesapce ( unlsess you've changed the default prefix ).
+			// Will be automatically prefixed with the component namespace by default.
+			return {FILE_NAME_WITHOUT_EXTENSION};
+		}
+	
+		getPrefix() {
+			// Override the default prefix.
+			return {CUSTOM_PREFIX_KEBAB_CASE};
+		}
+	
+		getScopes() {
+			// List of `HTMLElement`s that the state will dispatch events && add CSS classes to.
+			// ( Defaults to `document.body` )
+			return [
+				window.document.body,
+			];
+		}
+	
+		getOptions() {
+			// Object of options that the state can be set to, with an optional callback for each option.
+			return {
+				{OPTION_VALUE_KEBAB_CASE}: '',
+				'on': ( oldValue, newValue ) => {
+					// Callback that runs when the state is set to `on`.
+				},
+				'off': this.onStateOff,
+			};
+		}
+ 
+		onChange( oldValue, newValue ) {
+			// Callback that runs on state change.
+			console.log( { oldValue, newValue } );
+		}
+	}
+	```
+
+  > Legend
```

Comment:
What does it mean?

Reply:
https://translate.google.com/?sl=iw&tl=en&text=%D7%9E%D6%B4%D7%A7%D7%A8%D6%B8%D7%90&op=translate

See also @iNewLegend's docs, it's the same as there.